### PR TITLE
Add in support for murmur3 hashing of structs

### LINKED ISSUE #1920: [FEA] Support murmur3 hashing of structs
We want to eventually join on structs and to do efficiently we need to be able to hash partition on structs which if we want to be compatible with spark requires us to support murmur3 hashing of them.

### PR DESCRIPTION
**Pull request by revans2**
This depends on https://github.com/rapidsai/cudf/pull/7714

This fixes  #1920

### CODE REVIEW

**Review by jlowe — APPROVED** [review]

**Review by gerashegalov — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by jlowe** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala [hunk 1]
```diff
@@ -2334,7 +2334,7 @@ object GpuOverrides {
       "Murmur3 hash operator",
       ExprChecks.projectNotLambda(TypeSig.INT, TypeSig.INT,
         repeatingParamCheck = Some(RepeatingParamCheck("input",
-          TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL,
+          (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.STRUCT).nested(),
```
[the remark this answers is not in the document]
Comment (revans2) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:2346]: The test I added in repart_test.py is for a struct of a struct.

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (gerashegalov) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:2346]: missed that

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**